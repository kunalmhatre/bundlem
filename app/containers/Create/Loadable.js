/**
 *
 * Asynchronously loads the component for Create
 *
 */

import loadable from 'loadable-components';

export default loadable(() => import('./index'));
