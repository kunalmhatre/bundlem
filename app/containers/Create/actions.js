/*
 *
 * Create actions
 *
 */

import CREATE_BUNDLE from './constants';

export default function createBundleAction(title, description) {
  return {
    title,
    description,
    type: CREATE_BUNDLE,
  };
}
