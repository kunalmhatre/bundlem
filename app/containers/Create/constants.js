/*
 *
 * Create constants
 *
 */

const CREATE_BUNDLE = 'app/Create/CREATE_BUNDLE';

export default CREATE_BUNDLE;
