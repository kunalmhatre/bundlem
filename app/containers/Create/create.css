.create-title {
    font-size: 1.7rem;
    font-weight: 700;
    padding: 3rem 0 3rem 0;
}

.create-button {
    margin-top: 1rem;
    float: right;
}

.create-bundle-card {
    display: flex;
    justify-content: center;
    margin-top: 10rem;
}

@media only screen and (max-width: 991px) {
    .create-title {
        font-size: 1.5rem;
    }
    .create-bundle-card {
        display: none;
    }
}
