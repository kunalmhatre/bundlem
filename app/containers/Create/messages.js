/*
 * Create Messages
 *
 * This contains all the text for the Create container.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.containers.Create';

export default defineMessages({
  createTitle: {
    id: `${scope}.createTitle`,
    defaultMessage: 'Set a title for your Bundle',
  },
  titleLabel: {
    id: `${scope}.titleLabel`,
    defaultMessage: 'TITLE',
  },
  titlePlaceholder: {
    id: `${scope}.titlePlaceholder`,
    defaultMessage: 'Enter title',
  },
  descriptionLabel: {
    id: `${scope}.descriptionLabel`,
    defaultMessage: 'DESCRIPTION',
  },
  descriptionPlaceholder: {
    id: `${scope}.descriptionPlaceholder`,
    defaultMessage: 'Enter more description (Optional)',
  },
  nextButton: {
    id: `${scope}.nextButton`,
    defaultMessage: 'NEXT',
  },
  errorTitleEmpty: {
    id: `${scope}.errorTitleEmpty`,
    defaultMessage: 'Please enter the title',
  },
  errorTitleLength: {
    id: `${scope}.errorTitleLength`,
    defaultMessage: 'Exceeds maximum length of 120 characters',
  },
  errorDescriptionLength: {
    id: `${scope}.errorDescriptionLength`,
    defaultMessage: 'Exceeds maximum length of 320 characters',
  },
  helmetTitle: {
    id: `${scope}.helmetTitle`,
    defaultMessage: 'Create',
  },
  helmetDescription: {
    id: `${scope}.helmetDescription`,
    defaultMessage: 'Create your bundle of online resources.',
  },
});
