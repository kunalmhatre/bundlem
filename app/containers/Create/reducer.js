/*
 *
 * Create reducer
 *
 */

import { fromJS } from 'immutable';
import CREATE_BUNDLE from './constants';

export const initialState = fromJS({
  title: '',
  description: '',
});

function createReducer(state = initialState, action) {
  switch (action.type) {
    case CREATE_BUNDLE:
      return state
        .set('title', action.title)
        .set('description', action.description);
    default:
      return state;
  }
}

export default createReducer;
