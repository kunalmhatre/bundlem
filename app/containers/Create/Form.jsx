import React from 'react';
import { withFormik } from 'formik';
import PropTypes from 'prop-types';
import * as yup from 'yup';
import { Row, Col } from 'antd';
import { FormattedMessage } from 'react-intl';

import messages from './messages';
import BundleCard from '../../components/BundleCard';
import Button from '../../components/Button';
import Input from '../../components/Input';

const formikValuesPropType = PropTypes.shape({
  title: PropTypes.string,
  description: PropTypes.string,
});

const propTypes = {
  values: formikValuesPropType.isRequired,
  errors: formikValuesPropType.isRequired,
  touched: PropTypes.shape({
    title: PropTypes.bool,
    description: PropTypes.bool,
  }).isRequired,
  handleChange: PropTypes.func.isRequired,
  handleBlur: PropTypes.func.isRequired,
  isValid: PropTypes.bool.isRequired,
  onSubmit: PropTypes.func,
};

const defaultProps = {
  onSubmit: () => null,
};

/**
 * Form for Create page
 * @param {object} values - BY FORMIK HOC (contains values for the input fields)
 * @param {object} errors - BY FORMIK HOC (contains errors for the input fields)
 * @param {object} touched - BY FORMIK HOC (states whether specific input field was touched)
 * @param {function} handleChange - BY FORMIK HOC (on change handler function)
 * @param {function} handleBlur - BY FORMIK HOC (on blur handler function)
 * @param {boolean} isValid - BY FORMIK HOC (states whether form is valid for submission)
 * @param {function} onSubmit - On submit handler function
 */
function BaseForm({
  values,
  errors,
  touched,
  handleChange,
  handleBlur,
  isValid,
  onSubmit,
}) {
  const responsiveForm = {
    xs: 24,
    sm: 24,
    md: 24,
    lg: 10,
    xl: 10,
    xxl: 10,
  };

  return (
    <Row>
      <Col {...responsiveForm}>
        <div className="create-title">
          <FormattedMessage {...messages.createTitle} />
        </div>
        <form>
          <Input
            autoFocus
            id="title"
            name="title"
            label={messages.titleLabel}
            placeholder={messages.titlePlaceholder}
            onChange={handleChange}
            onBlur={handleBlur}
            value={values.title}
            showError={!!(touched.title && errors.title)}
            errorMessage={messages[errors.title]}
          />
          <Input
            id="description"
            name="description"
            label={messages.descriptionLabel}
            placeholder={messages.descriptionPlaceholder}
            onChange={handleChange}
            onBlur={handleBlur}
            value={values.description}
            showError={!!(errors.description)}
            errorMessage={messages[errors.description]}
          />
          <Button
            className="create-button"
            text={messages.nextButton}
            onClick={() => onSubmit(values.title.trim(), values.description.trim())}
            disabled={!isValid}
          />
        </form>
      </Col>
      <Col
        className="create-bundle-card"
        span={14}
      >
        <BundleCard
          bundleTitle={values.title || 'Bundle'}
          resourcesCount={0}
          clickable={false}
        />
      </Col>
    </Row>
  );
}

BaseForm.propTypes = propTypes;
BaseForm.defaultProps = defaultProps;

const Form = withFormik({
  mapPropsToValues({ title, description }) {
    return {
      title: title || '',
      description: description || '',
    };
  },
  validationSchema: yup.object().shape({
    title: yup
      .string()
      .trim()
      .max(120, 'errorTitleLength')
      .required('errorTitleEmpty'),
    description: yup
      .string()
      .trim()
      .max(320, 'errorDescriptionLength'),
  }),
  isInitialValid({ title }) {
    return !!(title);
  },
})(BaseForm);

export default Form;
export { BaseForm };
