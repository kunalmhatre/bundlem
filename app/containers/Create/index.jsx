import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Helmet } from 'react-helmet';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import Form from './Form';
import PageLayout from '../../components/PageLayout';
import injectReducer from '../../utils/injectReducer';
import reducer from './reducer';
import createBundleAction from './actions';
import { makeSelectTitle, makeSelectDescription } from './selectors';
import './create.css';

const propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  createBundle: PropTypes.func.isRequired,
};

/**
 * Create page
 * @param {string} title - Title (if previously present) from state
 * @param {string} description - Description (if previously present) from state
 * @param {function} createBundle - Redux action to save title and description in the state
 */
function Create({ title, description, createBundle }) {
  return (
    <React.Fragment>
      <Helmet>
        <title>Create</title>
        <meta name="description" content="Description of Create" />
      </Helmet>
      <PageLayout>
        <Form
          title={title}
          description={description}
          onSubmit={createBundle}
        />
      </PageLayout>
    </React.Fragment>
  );
}

Create.propTypes = propTypes;

const mapStateToProps = createStructuredSelector({
  title: makeSelectTitle(),
  description: makeSelectDescription(),
});

function mapDispatchToProps(dispatch) {
  return {
    createBundle: (title, description) => dispatch(createBundleAction(title, description)),
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'create', reducer });

export default compose(
  withReducer,
  withConnect,
)(Create);

export { mapDispatchToProps };
