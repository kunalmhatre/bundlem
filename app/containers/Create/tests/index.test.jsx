import React from 'react';
import { Provider } from 'react-redux';
import createComponentWithIntl from '../../../../internals/testing/createComponentWithIntl';
import history from '../../../utils/history';
import configureStore from '../../../configureStore';
import Create, { mapDispatchToProps } from '../index';
import createBundleAction from '../actions';

describe('<Create />', () => {
  const store = configureStore({}, history);

  it('renders successfully', () => {
    expect(
      createComponentWithIntl(
        <Provider store={store}>
          <Create />
        </Provider>,
      ).toJSON(),
    ).toMatchSnapshot();
  });

  it('dispatches createBundleAction', () => {
    const dispatch = jest.fn();
    const props = mapDispatchToProps(dispatch);
    const title = 'testTitle';
    const description = 'testDescription';

    props.createBundle(title, description);
    expect(dispatch).toHaveBeenCalledWith(createBundleAction(title, description));
  });
});
