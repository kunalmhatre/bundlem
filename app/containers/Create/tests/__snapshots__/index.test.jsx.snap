// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Create /> renders successfully 1`] = `
<div
  className="ant-row page-layout theme-blue"
>
  <div
    className="ant-col-22 ant-col-offset-1"
  >
    <div
      className="page-layout-header-content"
    >
      <header
        className="header theme-blue"
      >
        <span>
          Bundlem
        </span>
      </header>
      <section
        className="page-layout-content"
      >
        <div
          className="ant-row"
        >
          <div
            className="ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-10 ant-col-xl-10 ant-col-xxl-10"
          >
            <div
              className="create-title"
            >
              <span>
                Set a title for your Bundle
              </span>
            </div>
            <form>
              <div
                className="input theme-blue"
              >
                <label
                  htmlFor="title"
                >
                  <div
                    className="input-label"
                  >
                    <span>
                      TITLE
                    </span>
                  </div>
                  <input
                    autoFocus={true}
                    className="input-field"
                    id="title"
                    name="title"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Enter title"
                    value=""
                  />
                </label>
                <div
                  className="input-error-message theme-blue-error-message"
                />
              </div>
              <div
                className="input theme-blue"
              >
                <label
                  htmlFor="description"
                >
                  <div
                    className="input-label"
                  >
                    <span>
                      DESCRIPTION
                    </span>
                  </div>
                  <input
                    autoFocus={false}
                    className="input-field"
                    id="description"
                    name="description"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Enter more description (Optional)"
                    value=""
                  />
                </label>
                <div
                  className="input-error-message theme-blue-error-message"
                />
              </div>
              <button
                className="button theme-blue-inverse theme-blue-border create-button"
                disabled={true}
                onClick={[Function]}
                type="button"
              >
                <span>
                  NEXT
                </span>
              </button>
            </form>
          </div>
          <div
            className="ant-col-14 create-bundle-card"
          >
            <div
              className="bundle-card theme-blue-inverse"
            >
              <div
                className="bundle-card-title"
              >
                <a
                  className="theme-blue-inverse"
                  href={null}
                >
                  Bundle
                </a>
              </div>
              <div
                className="bundle-card-actions"
              >
                <div>
                  <span>
                    0
                  </span>
                  <span>
                     Res.
                  </span>
                </div>
                <div>
                  <a
                    href={null}
                  >
                    <span>
                      view
                    </span>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
    <footer
      className="footer theme-blue"
    >
      <a
        className="footer-link theme-blue"
        href="https://github.com/kunalmhatre/bundlem/"
      >
        <span>
          source
        </span>
      </a>
      <a
        className="footer-link theme-blue"
        href="https://github.com/kunalmhatre/bundlem/issues/new/choose"
      >
        <span>
          REPORT ISSUE
        </span>
      </a>
      <a
        className="footer-link theme-blue"
        href="https://kunalmhatre.com/bundlem/"
      >
        <span>
          STORY
        </span>
      </a>
      <a
        className="footer-link theme-blue"
        href="https://kunalmhatre.com/"
      >
        <span>
          AUTHOR
        </span>
      </a>
    </footer>
  </div>
</div>
`;
